app/services/utils/service_test.go

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2022 LimeChain Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/limechain/hedera-eth-bridge-validator/app/clients/evm/contracts/router"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/client"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/service"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	"github.com/limechain/hedera-eth-bridge-validator/test/mocks"
+	"github.com/stretchr/testify/assert"
+	"strings"
+	"testing"
+)
+
+var (
+	svc              *utilsService
+	lockHash         common.Hash
+	burnHash         common.Hash
+	burnErc721       common.Hash
+	someHash         common.Hash
+	evmTx            = "0xa83be7d95c58f57e11f5c27dedd963217d47bdeab897bc98f2f5410d9f6c0026"
+	evmTxHash        = common.HexToHash(evmTx)
+	expectedBridgeTx = "0.0.1-123-123"
+	expectedResult   = &service.BridgeTxId{
+		BridgeTxId: expectedBridgeTx,
+	}
+)
+
+func setup() {
+	mocks.Setup()
+
+	svc = &utilsService{
+		evmClients: map[uint64]client.EVM{
+			80001: mocks.MEVMClient,
+		},
+		burnEvt: mocks.MBurnService,
+		log:     config.GetLoggerFor("Utils Service"),
+	}
+
+	routerAbi, _ := abi.JSON(strings.NewReader(router.RouterABI))
+	burnHash = routerAbi.Events["Burn"].ID
+	lockHash = routerAbi.Events["Lock"].ID
+	burnErc721 = routerAbi.Events["BurnERC721"].ID
+}
+
+func Test_New(t *testing.T) {
+	setup()
+
+	actual := New(map[uint64]client.EVM{
+		80001: mocks.MEVMClient,
+	}, mocks.MBurnService)
+
+	assert.Equal(t, svc, actual)
+}
+
+func Test_ConvertEvmTxIdToHederaTxId_LockEvent(t *testing.T) {
+	setup()
+	mocks.MEVMClient.On("WaitForTransactionReceipt", evmTxHash).Return(&types.Receipt{
+		Logs: []*types.Log{
```

Comment:
maybe we can have it at the beginning `var (...)` and re-used in the `Test_ConvertEvmTxIdToHederaTxId_LockEvent`, `Test_ConvertEvmTxIdToHederaTxId_BurnEvent`, and `Test_ConvertEvmTxIdToHederaTxId_BurnErc721Event`?